scrapy_time_machine/storages.py

```diff
@@ -97,3 +98,51 @@ def _read_data(self, spider, request):
 
     def _request_key(self, request):
         return request_fingerprint(request)
+
+
+class S3TimeMachineStorage(DbmTimeMachineStorage):
+    def __init__(self, settings):
+        self.s3 = settings.get("TIME_MACHINE_URI")
+        self.s3_client = boto3.client(
+            "s3",
+            aws_access_key_id=settings.get("AWS_ACCESS_KEY_ID"),
+            aws_secret_access_key=settings.get("AWS_SECRET_ACCESS_KEY"),
+        )
+
+    def open_spider(self, spider):
+        self._spider = spider
+
+        self._prepare_time_machine()
+        self.snapshot_uri = self.s3
+
+        if not self.snapshot_uri:
+            raise CloseSpider("Snapshot uri not configured.")
+
+        self.db = open(self.local_path_to_db, "c")
+
+        logger.debug(
+            "Using S3 time machine storage in %(dbpath)s"
+            % {"dbpath": self.snapshot_uri},
+            extra={"spider": spider},
+        )
+
+    def _prepare_time_machine(self, settings):
+        if settings.get("TIME_MACHINE_RETRIEVE"):
+            # download db file from s3 snapshot_uri
+            compressed_db_file = self.s3_client.download_file('compressed_db_file', self.snapshot_uri)
```

Comment:
You are hardcoding the AWS bucket value here to `compressed_db_file`   and you are missing the local field path.

```
S3.Client.download_file(Bucket, Key, Filename, ExtraArgs=None, Callback=None, Config=None)
```
You need to parse the `snapshot_uri` to get bucket (`netloc`) and the key (`path`). 

For the local field path, you can use `tempfile.mkstemp` to get the Path to temporal file or the same approach you used with NamedTemporaryFile below